### ISSUE TITLE
ValueError: export_outputs must be a dict error when saving model_to_estimator

### ISSUE THREAD

**Issue by dtsukiyama** [awslabs/amazon-sagemaker-examples#189]
Hi I have been trying to test ```tf.keras.estimator.model_to_estimator(keras_model=model)``` and save it i order to set up hosting for the model like in this example: https://github.com/awslabs/amazon-sagemaker-examples/blob/master/advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb

However I keep receiving this error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-75-a22213ebd37e> in <module>()
      1 exported_model = model.export_savedmodel(export_dir_base = 'export/Servo/', 
----> 2                                serving_input_receiver_fn = serving_input_fn)

~/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py in export_savedmodel(self, export_dir_base, serving_input_receiver_fn, assets_extra, as_text, checkpoint_path)
    515           serving_input_receiver.receiver_tensors,
    516           estimator_spec.export_outputs,
--> 517           serving_input_receiver.receiver_tensors_alternatives)
    518 
    519       if not checkpoint_path:

~/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/tensorflow/python/estimator/export/export.py in build_all_signature_defs(receiver_tensors, export_outputs, receiver_tensors_alternatives)
    191     receiver_tensors = {_SINGLE_RECEIVER_DEFAULT_NAME: receiver_tensors}
    192   if export_outputs is None or not isinstance(export_outputs, dict):
--> 193     raise ValueError('export_outputs must be a dict.')
    194 
    195   signature_def_map = {}

ValueError: export_outputs must be a dict.


```

My code:

```python

import numpy as np
import os
import tensorflow as tf
from sklearn.preprocessing import LabelEncoder
from sklearn.externals import joblib


def featureTransform(features, max_words):
    tokenize = tf.keras.preprocessing.text.Tokenizer(num_words=max_words, char_level=False)
    tokenize.fit_on_texts(features) 
    return tokenize.texts_to_matrix(features).astype(np.float32) 

def encodeLabels(labels):
    encoder = LabelEncoder()
    encoder.fit(labels)
    y = encoder.transform(labels)
    num_classes = np.max(y) + 1
    print("num classes: {}".format(num_classes))
    return tf.keras.utils.to_categorical(y, num_classes).astype(np.float32)

def baselineModel(max_words, num_classes):
    model = tf.keras.models.Sequential()
    model.add(tf.keras.layers.Dense(500, activation='relu', input_shape=(max_words,), name="features"))
    model.add(tf.keras.layers.Dropout(0.5))
    model.add(tf.keras.layers.Dense(500, activation='relu'))
    model.add(tf.keras.layers.Dropout(0.5))
    model.add(tf.keras.layers.Dense(num_classes, activation='softmax'))
    model.compile(loss='categorical_crossentropy',
              optimizer='rmsprop',
              metrics=['accuracy'])
    
    return tf.keras.estimator.model_to_estimator(keras_model=model)


def serving_input_fn():
    feature_spec = {'features_input': tf.FixedLenFeature(dtype=tf.float32, shape=[500])}
    return tf.estimator.export.build_parsing_serving_input_receiver_fn(feature_spec)()


def train_input_fn(training_dir, params):
    """Returns input function that would feed the model during training"""
    return input_function(training_dir, 'assignment_train.csv')

def input_function(training_dir, training_filename, shuffle=False):
    training_set = tf.contrib.learn.datasets.base.load_csv_without_header(
    filename=os.path.join(training_dir, training_filename), target_dtype=np.str, features_dtype=np.float32)
    
    input_fn = tf.estimator.inputs.numpy_input_fn(
        x={"features_input":  np.array(training_set.data)}, 
        y=encodeLabels(training_set.target),
        num_epochs=100,
        shuffle=shuffle
    )
    return input_fn

```

I run the following:
```python
model = baselineModel(500, 123)
model.train(input_fn=input_function('data','assignment_train.csv', shuffle=True))
score = model.evaluate(input_function('data','assignment_train.csv', shuffle=True), steps = 100)
exported_model = model.export_savedmodel(export_dir_base = 'export/Servo/', 
                               serving_input_receiver_fn = serving_input_fn)
```

Any insight would be greatly appreciated!

**Comment by mvsusp** [comment]
Hi @dtsukiyama,

Thanks for using SageMaker!

The error `ValueError: export_outputs must be a dict` is a known [bug on TensorFlow 1.4, and it was fixed in TensorFlow 1.5](https://github.com/tensorflow/tensorflow/issues/14284).

If you update your local version to TF 1.5, the issue will be fixed. 
The current version of TensorFlow serving in the SageMaker TensorFlow container is 1.4 (we are in the process to release the new version) but I don't see incompatibilities in your model between both versions.

I will close this question. Feel free to open it again if you have more questions.

Best,
Marcio

**Comment by dtsukiyama** [comment]
Thanks @mvsusp !
